cloudify_agent/api/pm/detach.py

```diff
@@ -64,11 +64,62 @@ def before_self_stop(self):
         self._runner.run(self.create_disable_cron_script())
         super(DetachedDaemon, self).before_self_stop()
 
+    def _clean_queue(self, client):
+        queue = '{0}_service'.format(self.queue)
+        self._logger.debug('Cleaning amqp agent queue {0}'.format(queue))
+        with client:
+            try:
+                client.channel_method(
+                    'queue_delete',
+                    queue=queue,
+                    if_empty=True,
+                    wait=False
+                )
+            except Exception as err:
+                self._logger.debug(
+                    'Either queue {0} does not exist or '
+                    'something else happened {1}'.format(queue, str(err)))
+            else:
+                self._logger.debug(
+                    'AMQP agent queue {0} deleted '
+                    'successfully'.format(queue)
+                )
+
+    def _clean_exchange(self, client):
+        self._logger.info(
+            'Cleaning amqp agent'
+            ' exchange {0}'.format(self.queue)
+        )
+        with client:
+            try:
+                client.channel_method(
+                    'exchange_delete',
+                    exchange=self.queue,
+                )
+            except Exception as err:
+                self._logger.debug(
+                    'Either exchange {0} does not exist or '
+                    'something else happened {1}'.format(self.queue, str(err)))
```

Comment:
"something else happened" lol!
```suggestion
                    'Error deleting exchange %s: %s', self.queue, err)
```